java/org/apache/tomcat/util/net/AbstractEndpoint.java

```diff
@@ -1312,6 +1340,30 @@ public final void closeServerSocketGraceful() {
     }
 
 
+    /**
+     * Wait for the client connections to the server to close gracefully. The
+     * method will return when all of the client connections have closed or the
+     * method has been waiting for {@code waitTimeMillis}.
+     *
+     * @param waitMillis    The maximum time to wait in milliseconds for the
+     *                      client connections to close.
+     *
+     * @return The wait time, if any remaining when the method returned
+     */
+    public final long closeConnectionsAwait(long waitMillis) {
```

Comment:
Maybe the method name should be `awaitConnectionsClose` ?
Now it sounds like it closes and waits, but actually it just waits.

Reply:
Good suggestion. Fixed.